File: ecmascript/bob/bob.js
class Bob {
  hey(message) {
    const text = message.trim();

    if (text === text.toUpperCase() && text !== text.toLowerCase()) {
      return 'Whoa, chill out!';
    } else if (text.endsWith('?')) {
      return 'Sure.';
    } else if (text === '') {
      return 'Fine. Be that way!';
    }

    return 'Whatever.';
  }
}

export default Bob;

File: scala/bob/src/main/scala/Bob.scala
class Bob {
  def hey(message: String): String = {
    if (message.trim.isEmpty) {
      "Fine. Be that way!"
    } else if (message == message.toUpperCase && message != message.toLowerCase) {
      "Whoa, chill out!"
    } else if (message.last == '?') {
      "Sure."
    } else {
      "Whatever."
    }
  }
}
